test/units/module_utils/common/process/test_get_bin_path.py

```diff
@@ -27,6 +27,24 @@ def test_get_bin_path(mocker):
     assert '/usr/local/bin/notacommand' == get_bin_path('notacommand')
 
 
+def test_get_bin_path_usr_local_bin_not_in_env(mocker):
+    path = '/usr/local/sbin:/usr/sbin:/usr/bin:/sbin:/bin'
+    mocker.patch.dict('os.environ', {'PATH': path})
+    mocker.patch('os.pathsep', ':')
+
+    mocker.patch('os.path.isdir', return_value=False)
+    mocker.patch('ansible.module_utils.common.process.is_executable', return_value=True)
+
+    # pytest-mock 2.0.0 will throw when os.path.exists is messed with
+    # and then another method is patched afterwards. Likely
+    # something in the pytest-mock chain uses os.path.exists internally, and
+    # since pytest-mock prohibits context-specific patching, there's not a
+    # good solution. For now, just patch os.path.exists last.
+    mocker.patch('os.path.exists', side_effect=[False, True])
```

Comment:
This looks like the cause of the test failure. Rather than using side_effect with a list of values, might be easier to read like this:

```python
    def mock_path_exists(path):
        return path in ('/usr/local/bin', '/usr/local/bin/notacommand')

    mocker.patch('os.path.exists', return_value=mock_path_exists)
```